js/gestionPresupuesto.js

```diff
@@ -1,18 +1,50 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
 
-// TODO: Variable global
+// Variable global
+let presupuesto = 0;
 
-
-function actualizarPresupuesto() {
+function actualizarPresupuesto(actPresupuesto) {
     // TODO
+     
+    if ( actPresupuesto >= 0)
+    {
+        presupuesto = actPresupuesto;
+        return presupuesto;
+    }
+    else {
+        //console.error('dato novalido');
+        return -1;
+    }
 }
 
+
 function mostrarPresupuesto() {
     // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`;
 }
 
-function CrearGasto() {
+function CrearGasto(descripcion, valor) {
     // TODO
+
+  
+   let gasto = {
```

Comment:
No está mal, aunque se pedía hacerlo con un constructor.

Las funciones constructoras se llaman con `new`, por lo que devuelven un objeto sin necesidad de crearlo, como tú has hecho. Referencia: https://es.javascript.info/constructor-new

Básicamente tienes que quitar la creación del literal del objeto (`let gasto = {...}`) y dejar solo las referencias a sus propiedades mediante `this`. Intenta hacerlo de esa manera y cuando lo tengas vuelve a abrir la PR.